src/hist/basehist.py

```diff
@@ -320,7 +325,21 @@ def plot1d(
 
         import hist.plot
 
-        return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        if self.ndim == 1:
+            return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        else:
+            if overlay is not None:
+                (cat_ax,) = [ax for ax in self.axes if ax.name == overlay]
```

Comment:
```suggestion
            if overlay is not None:
                cat_ax = self.axes[overlay]
```

Will also support integer specification.